drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/handlers/TitledAttachmentFileWidget.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.handlers;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.gwt.core.client.GWT;
+import com.google.gwt.user.client.ui.Composite;
+import com.google.gwt.user.client.ui.VerticalPanel;
+import org.guvnor.common.services.project.model.WorkspaceProject;
+import org.gwtbootstrap3.client.ui.FormLabel;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.kie.workbench.common.screens.examples.client.wizard.widgets.ComboBox;
+import org.kie.workbench.common.screens.explorer.model.FolderItemType;
+import org.kie.workbench.common.screens.library.api.AssetInfo;
+import org.kie.workbench.common.screens.library.api.AssetQueryResult;
+import org.kie.workbench.common.screens.library.api.ProjectAssetsQuery;
+import org.kie.workbench.common.screens.library.client.screens.assets.AssetQueryService;
+import org.kie.workbench.common.screens.library.client.util.LibraryPlaces;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.ext.widgets.common.client.callbacks.DefaultErrorCallback;
+import org.uberfire.util.URIUtil;
+
+public class TitledAttachmentFileWidget extends Composite {
+
+    protected VerticalPanel fields = GWT.create(VerticalPanel.class);
+    protected FormLabel titleLabel = GWT.create(FormLabel.class);
+    protected ComboBox comboBox = GWT.create(ComboBox.class);
+    protected LibraryPlaces libraryPlaces;
+    protected AssetQueryService assetQueryService;
+    protected WorkspaceProject workspaceProject;
+
+    public TitledAttachmentFileWidget() {
+        this("", null, null);
```

Comment:
@gitgabrio, @manstis 
Can anybody explain why this default constructor is there and use call of the other constructor with pretty much useless arguments - empty string and nulls?

we have migrating static analysis from findbugs to spotbugs and all suspicious workarounds should be removed like this reasonable error https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html?highlight=NP_NULL_PARAM_DEREF_NONVIRTUAL so the explanation would be welcome here.

Reply:
@mareknovotny 
Not remember exactly why I put it there, is not needed anymore. I've removed it on this PR https://github.com/kiegroup/drools-wb/pull/1079.
Beside that, this component is already to be completely deleted:
https://issues.jboss.org/browse/DROOLS-3420

Reply:
great, so we can remove the filter for NPE in spotbugs-exclusion.xml!
@mbiarnes  fyi this reduces the need of the spotbugs filter in https://github.com/kiegroup/drools-wb/pull/1058

Reply:
I see a lot of that type of code in Stunner.

It's usually where the properties of a class are defined as `final` and an instance created by CDI.. the _default_ zero parameter constructor _may_ be invoked by CDI when it has to create a proxy for circular injection. CDI must do some _Reflective_ voodoo to later set the value of the `final` properties.

Reply:
@manstis 
Yes, exactly. I'm pretty sure I add this for that reason -I like to have "final" properties wherever possible. But then, probably, I had to remove the "final" (testing ?), and I forgot to remove that constructor.
Anyway, I run the code without that empty constructor, and also all the tests, and it seems it is not needed.

Reply:
@manstis sure if that is CDI proxy bean it requires the default constructor and whenever final is put, but setting it to null and be null forever doesn't make sense to me from natural POV.